src/services/ui/src/hooks/useSyncStatus.ts

```diff
@@ -0,0 +1,73 @@
+import { getItem, useGetItem } from "@/api";
+import { Route, urlEmbedQuery } from "@/components/Routing";
+import { useNavigate } from "react-router-dom";
+import { useQuery } from "@tanstack/react-query";
+import { useRef, useState } from "react";
+import { type SEATOOL_STATUS } from "shared-types";
+import { queryClient } from "@/router";
+import { ItemResult } from "shared-types/opensearch/main";
+import { OsUrlState } from "@/components";
+import { authorityById } from "@/utils";
+
+export type SeaStatus = (typeof SEATOOL_STATUS)[keyof typeof SEATOOL_STATUS];
+
+export const useSyncStatus = ({
+  path,
+  isCorrectStatus,
```

Comment:
I have a similar comment for this one.
5 times this hook is called, `SEATOOL_STATUS` is used and pulled into the component when its not used anywhere else. Additionally there are twice duplicates of a similar function like: ` return data._source.actionType === "New";`
I was wondering if for this second part of the hook if maybe we wanted to define a set of rules for that we use to test against?

Like:
```javascript
const StatusCheck = {
pendingRai: (data) => {
    return (
      data._source.seatoolStatus === SEATOOL_STATUS.PENDING_RAI &&
      !!data._source.raiRequestedDate
    );
  },
enabled: (data) => {
    const isEnabled = ACTION_WORD === "Enable";
    return data._source.raiWithdrawEnabled === isEnabled;
  },...}
```


Then the hook call could use the key from that map to choose which function to run, instead of passing the function in yourself.
  
  
This might make it more complicated then it needs to be. I trust your judgement to what you think will be best here. 😄